src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 && !resolved) {
+      clearTimeout(timeout);
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  const onClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+      document.removeEventListener('click', onClick);
+      document.removeEventListener('contextmenu', onClick);
+    }
```

Comment:
The same issue is here. You are creating a new function every time the `click` or `contextmenu` event is triggered. It would be better to define this function outside of the event listener and reference it.